sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -59,7 +59,10 @@ trait TimeZoneAwareExpression extends Expression {
   /** Returns a copy of this expression with the specified timeZoneId. */
   def withTimeZone(timeZoneId: String): TimeZoneAwareExpression
 
-  @transient lazy val zoneId: ZoneId = DateTimeUtils.getZoneId(timeZoneId.get)
+  @transient lazy val zoneId: ZoneId = timeZoneId match {
+    case Some(x) => DateTimeUtils.getZoneId(x)
+    case None => TimeZone.getDefault.toZoneId
```

Comment:
Don't think we should use the default JVM time zone. At least, `spark.sql.session.timeZone`.